Add node pool

ograu opened the pull request:
`AddNodePool` component is basically copy/pasted from CreateCluster component and adapted to its needs: I removed everything related to Azure and KVM because AFAIK node pools aren't supported in Azure and KVM by now. Correct me if I'm wrong, please.

Also in this PR: 

- I've started to use Immer to update nested properties on local state object
- I've started to add the HTML element name as a suffix in emotion styled variables. I think Marian does it as well (?). It makes JSX in render easier to read IMO.

Form view:

<img width="913" alt="Captura de pantalla 2019-09-25 a las 13 10 36" src="https://user-images.githubusercontent.com/14203438/65595888-072d2b80-df96-11e9-9c07-c2696cb35935.png">


With validation errors/messages (some adjustments to be made on margins):

<img width="912" alt="Captura de pantalla 2019-09-25 a las 13 11 30" src="https://user-images.githubusercontent.com/14203438/65595894-0bf1df80-df96-11e9-8eff-b5e970ff6e82.png">

ograu commented:
@marian "Add Node Pool" button excludes the form and the other way around, I mean when the form pops up, the button is hidden. This is not what I see in the wireframes but if I follow what we have there, the user could have more than one form in her/his screen and... I don't like it. Let's discuss it if you think otherwise.

<img width="821" alt="Captura de pantalla 2019-09-25 a las 15 11 33" src="https://user-images.githubusercontent.com/14203438/65603909-cf7aaf80-dfa6-11e9-8515-9373d5122eeb.png">

oponder commented on src/components/cluster/new/aws_instance_type_selector.js:
```diff
@@ -103,7 +103,7 @@ class AWSInstanceTypeSelector extends React.Component {
   render() {
     return (
-      <div className='col-4'>
+      <div style={{ maxWidth: '200px' }}>
```
There might be a good reason in this case, but generally, would like to avoid inline styles.

oponder commented on src/reducers/clusterReducer.js:
```diff
@@ -134,6 +134,10 @@ const clusterReducer = produce((draft, action) => {
       return;
+
+    case types.NODEPOOL_CREATE_SUCCESS:
+      draft.items[action.clusterId].push(action.nodePool.id);
```
not sure about the API.. does it make sense to push the nodepool into the clusters list?

ograu replied:
It is actually an error, it has to be pushed inside the `nodePools`array of a cluster. 
Didn't noticed it because endpoint is not working and so this code has never been reached.
Thanks!

ograu commented:
ping @marians 

Now the form can switch from availability zones picker:

<img width="658" alt="Captura de pantalla 2019-10-01 a las 10 22 26" src="https://user-images.githubusercontent.com/14203438/65945967-64f9c180-e435-11e9-9593-190eebd34cea.png">

To availability zones labels:

<img width="543" alt="Captura de pantalla 2019-10-01 a las 10 23 03" src="https://user-images.githubusercontent.com/14203438/65946018-79d65500-e435-11e9-80a3-a750e8df3c77.png">



In order to have _clickable_ labels for availability zones I have:

- Adapted `AvailabilityZonesLabel` and `AvailabilityZonesLabels` components to accept checked/unchecked state
- Created a wrapper for `AvailabilityZonesLabels` responsible for passing the state in the correct shape to the form

This is how labels looks like when they are checked (A, B, C) and unchecked (D, E):

<img width="459" alt="Captura de pantalla 2019-10-01 a las 10 19 39" src="https://user-images.githubusercontent.com/14203438/65945842-1e0bcc00-e435-11e9-80f2-f374828bcead.png">

ograu commented:
- Changed default values for scaling, 
- Maximum availability zones is 4 now
- Improved validation and messages to the user for labels


<img width="529" alt="Captura de pantalla 2019-10-01 a las 12 53 44" src="https://user-images.githubusercontent.com/14203438/65956600-093a3300-e44b-11e9-8ac4-71bbf7f92055.png">

<img width="613" alt="Captura de pantalla 2019-10-01 a las 13 02 01" src="https://user-images.githubusercontent.com/14203438/65956883-b614b000-e44b-11e9-8613-5443b064c0dd.png">

ograu commented:
ping @marians Just realized that you weren't added here as a reviewer...

ograu commented:
In these lasts commits I made the modifications we've been discussing this morning:

- _Region:_ before region
- `0 nodes` when no the cluster has no node pools
- Kubernetes endpoint copyable
- Upgrade available underlined when hover
- Added text _To enable autoscaling, set minimum and maximum to different values._  below picker
- Changed background (too dark in my opinion)
- Disable text selection highlighting in all AZ labels

Things I couldn't change:

- Can't use the region instead AZ because in testing API we don't have the `datacenter` property on the object retrieved...

And finally, that weird `span`is something that `ReactCSSTransitionGroup` is adding 🤷‍♂ 

I will merge it and when we have the region will update it.
